src/scipp/core/data_group.py

```diff
@@ -248,6 +240,21 @@ def _data_group_binary(func: Callable, dg1: DataGroup, dg2: DataGroup, *args,
     })
 
 
+def _data_group_inplace(op: str, dg: DataGroup,
+                        other: Union[DataArray, Variable, numbers.Real]) -> DataGroup:
+    if isinstance(other, DataGroup):
+        if extra_keys := other.keys() - dg.keys():
+            raise ValueError(
+                "Cannot apply inplace operation because the right hand "
+                f"side has keys that are not in the left hand side: {extra_keys}")
```

Comment:
I think we should also raise when the rhs has missing keys, just like for `Dataset`?